packages/back-end/src/models/ReportModel.ts

```diff
@@ -57,7 +58,17 @@ const toInterface = (doc: ReportDocument): ReportInterface => {
         "_id",
       ]);
     default:
-      throw new Error("Invalid report type");
+      logger.error(
+        `Invalid report type: [${
+          (doc as ExperimentReportDocument)?.type ?? ""
+        }] (id: ${(doc as ExperimentReportDocument)?.id ?? "unknown"})`
+      );
+      return migrateExperimentReport(
+        omit(
+          (doc as ExperimentReportDocument).toJSON<ExperimentReportDocument>(),
+          ["__v", "_id"]
+        )
```

Comment:
Is migrating the report safe if it's an invalid report?

Reply:
yes, worst case scenario you can't load it depending on how broken the model is. If it's only a little bit broken, the migration should mostly work.